tests/utils/test_pydantic.py

```diff
@@ -64,3 +66,13 @@ def test_to_dot(input: str, expected: str):
 )
 def test_to_str(input: Any, expected: str):
     assert to_str(input) == expected
+
+
+def test_serialize_as_optional():
+    class Model(BaseModel):
+        foo: SerializeAsOptional[list[str]] = []
+
+    assert Model().model_dump() == {"foo": None}
+    assert Model().model_dump(exclude_defaults=True) == {}
+    assert Model().model_dump(exclude_unset=True) == {}
+    assert Model().model_dump(exclude_none=True) == {}
```

Comment:
If I understand the tests correctly, the field `foo` is only dumped as `"foo": None`. Will it ever be dumped as an empty list in this test-case?

Reply:
yep, that's exactly the point of the `SerializeAsOptional` type. If the value is falsy, i.e. collection is empty, it will be serialized to `None`